packages/elements/src/utils/oas/index.ts

```diff
@@ -26,11 +26,20 @@ const isOas3 = (parsed: unknown): parsed is OpenAPIObject =>
   'openapi' in parsed &&
   Number.parseFloat(String((parsed as Partial<{ openapi: unknown }>).openapi)) >= 3;
 
+const isOas31 = (parsed: unknown): parsed is OpenAPIObject =>
+  isObject(parsed) &&
+  'openapi' in parsed &&
+  Number.parseFloat(String((parsed as Partial<{ openapi: unknown }>).openapi)) === 3.1;
+
 const OAS_MODEL_REGEXP = /((definitions|components)\/?(schemas)?)\//;
 export const MODEL_REGEXP = /schemas\//;
 export const OPERATION_REGEXP = /\/operations\/.+|paths\/.+\/(get|post|put|patch|delete|head|options|trace)$/;
 
 export function transformOasToServiceNode(apiDescriptionDocument: unknown) {
+  if (isOas31(apiDescriptionDocument)) {
+    apiDescriptionDocument.jsonSchemaDialect = 'http://json-schema.org/draft-07/schema#';
+    return computeServiceNode(apiDescriptionDocument, oas3SourceMap, transformOas3Service, transformOas3Operation);
```

Comment:
This is a bit shady to me, as we now are introducing a side-effect to the function (and potentially to `API` as a whole - you can pass the spec object directly). Can I suggest
```suggestion
    return computeServiceNode({...apiDescriptionDocument, jsonSchemaDialect: 'http://json-schema.org/draft-07/schema#'}, oas3SourceMap, transformOas3Service, transformOas3Operation);
```
instead?

Reply:
Sure, that works as well